File: layers/src/test/java/com/psliusar/layers/MockedViewGroup.java
package com.psliusar.layers;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;

public class MockedViewGroup extends ViewGroup {

    private int childCount = 0;

    public MockedViewGroup(@NonNull Context context) {
        super(context);
    }

    @Override
    protected void onLayout(boolean changed, int l, int t, int r, int b) { }

    @Override
    public void setSaveFromParentEnabled(boolean enabled) { }

    @Override
    public int getChildCount() {
        return childCount;
    }

    @Override
    public void addView(@NonNull View child, int index) {
        childCount++;
    }

    @Override
    public void removeView(@NonNull View view) {
        childCount--;
    }
}

File: layers/src/test/java/com/psliusar/layers/MockedViewGroup.kt
package com.psliusar.layers

import android.content.Context
import android.view.View
import android.view.ViewGroup

class MockedViewGroup(context: Context) : ViewGroup(context) {

    private var childCount = 0

    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {}

    override fun setSaveFromParentEnabled(enabled: Boolean) {}

    override fun getChildCount(): Int = childCount

    override fun addView(child: View, index: Int) {
        childCount++
    }

    override fun removeView(view: View) {
        childCount--
    }
}
